src/AutoRest.CSharp/LowLevel/AutoRest/LowLevelTarget.cs

```diff
@@ -69,6 +68,31 @@ public static async Task ExecuteAsync(GeneratedCodeWorkspace project, InputNames
                 project.AddGeneratedFile($"{modelFactoryProvider.Type.Name}.cs", modelFactoryWriter.ToString());
             }
 
+            if (Configuration.GenerateTests)
+            {
+                // write test base and test env
+                var testBaseWriter = new ExpressionTypeProviderWriter(library.DpgTestBase);
+                testBaseWriter.Write();
+                project.AddGeneratedTestFile($"../../tests/Generated/Tests/{library.DpgTestBase.Type.Name}.cs", testBaseWriter.ToString());
```

Comment:
I would prefer not to override `ToString` method but pass `CodeWriter` instance as parameter to `ExpressionTypeProviderWriter` constuctor or `Write` method.

Reply:
Emmm any reason for that?
this is kind of a pattern in mgmt writers already introduced quite a time ago.
I could change it back, but we might need to align this

Reply:
Updated.